core/src/main/java/info/rmapproject/core/idservice/HttpArkIdService.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright 2017 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package info.rmapproject.core.idservice;
+
+import edu.ucsb.nceas.ezid.EZIDClient;
+import edu.ucsb.nceas.ezid.EZIDException;
+import org.h2.mvstore.MVMap;
+import org.h2.mvstore.MVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+public class HttpArkIdService implements IdService {
+
+    /** The log. */
+    private static final Logger log = LoggerFactory.getLogger(HttpUrlIdService.class);
+
+    /**  Wait time to retry when ID retrieval unsuccessful (5 seconds). */
+    private static final int RETRY_WAIT_TIME = 5000;
+
+    /** The HTTP ID service url. */
+    private String serviceUrl = "";
+
+    /** The ID prefix to be added. */
+    private String idPrefix = "";
+
+    /** Set a maximum retry attempts value if you want to cap the property setting. -1 means no maximum */
+    private int maxRetryAttempts = -1;
+
+    /** User name if using Basic Auth. */
+    private String userName = "";
+
+    /** User pwd if using Basic Auth. */
+    private String userPassword = "";
+
+    /** Length of ID to validate against. */
+    private int idLength = -1;
+
+    /** String regex to validate an ID against. */
+    private String idRegex = "";
+
+    /**file for persistent reserved ID storage **/
+    private String idStoreFile = "/tmp/idStoreFile";
```

Comment:
My concern here is use of a linux-specific path.  Can we use the `java.io.tmpdir` (or is it `java.io.tmpDir`) system property?